java/src/server/plugin/AbstractPlugin.java

```diff
@@ -0,0 +1,62 @@
+package server.plugin;
+
+import server.plugin.PluginConfig.PluginType;
+
+import java.net.URLClassLoader;
+import java.util.Map;
+
+/**
+ * Plugins are bundled into JAR files
+ */
+public abstract class AbstractPlugin {
```

Comment:
*Why*, exactly is an `AbstractPlugin` better than an `IPlugin`? The interface was seeming much more flexible before. If you do need duplicated code, you should have both. All of the things above should be storing and returning `IPlugin`, not `AbstractPlugin`. Our providers would then inherit from `AbstractPlugin`, where `AbstractPlugin` provides the storage and `IPlugin` represents the protocol that can be used and mocked. Also, everyone else's code that already uses `IPlugin` doesn't break all of a sudden.